examples/lines_bars_and_markers/timeline.py

```diff
@@ -0,0 +1,68 @@
+"""
+===============================================
+Creating a timeline with lines, dates, and text
+===============================================
+
+How to create a simple timeline using Matplotlib release dates.
+
+Timelines can be created with a collection of dates and text. In this example,
+we show how to create a simple timeline using the dates for recent releases
+of Matplotlib. First, we'll pull the data from GitHub.
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import urllib.request
+import json
+
+# Grab a list of Matplotlib releases
+url = 'https://api.github.com/repos/matplotlib/matplotlib/releases'
+data = json.loads(urllib.request.urlopen(url).read().decode())
+
+releases = []
+for irelease in data:
+    releases.append((irelease['tag_name'], irelease['published_at']))
+releases = pd.DataFrame(releases, columns=['name', 'date'])
+releases['date'] = pd.to_datetime(releases['date'])
+# Remove release candidates and betas
+releases = releases.loc[['rc' not in nm for nm in releases['name']]]
+releases = releases.loc[['b' not in nm for nm in releases['name']]]
+
+##############################################################################
+# Next, we'll iterate through each date and plot it on a horizontal line.
+# We'll add some styling to the text so that overlaps aren't as strong.
+#
+# Note that Matplotlib will automatically plot datetime inputs.
+
+levels = np.array([-5, 5, -3, 3, -1, 1])
+fig, ax = plt.subplots(figsize=(20, 5))
```

Comment:
The maximal figure width an example should have is 9 inch. See [this issue](https://github.com/matplotlib/matplotlib/issues/11190).